modules/core/src/test/scala/ch/epfl/bluebrain/nexus/rdf/EncoderSpec.scala

```diff
@@ -0,0 +1,205 @@
+package ch.epfl.bluebrain.nexus.rdf
+
+import java.time.{Instant, Period}
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+import cats.Contravariant
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.rdf.EncoderSpec.IntValue
+import ch.epfl.bluebrain.nexus.rdf.Graph.Triple
+import ch.epfl.bluebrain.nexus.rdf.Iri.{AbsoluteIri, Url}
+import ch.epfl.bluebrain.nexus.rdf.Node.{IriNode, Literal}
+import ch.epfl.bluebrain.nexus.rdf.Vocabulary.{rdf, schema}
+import ch.epfl.bluebrain.nexus.rdf.syntax.all._
+
+import scala.concurrent.duration.{Duration, FiniteDuration}
+
+class EncoderSpec extends RdfSpec {
+
+  private val example    = "http://example.com/"
+  private val exampleIri = Url.unsafe(example)
+
+  "An Encoder" should {
+    "contramap" in {
+      val encoder: Encoder[Int] = Contravariant[Encoder].contramap(Encoder[String])(_.toString)
+      val g                     = encoder(1)
+      g.node shouldEqual Literal("1")
+      g.triples.isEmpty shouldEqual true
+    }
+    "correctly encode primitives" in {
+      Encoder[Byte].apply(1.toByte) shouldEqual Graph(1.toByte)
+      Encoder[Short].apply(1.toShort) shouldEqual Graph(1.toShort)
+      Encoder[Int].apply(1) shouldEqual Graph(1)
+      Encoder[Long].apply(1L) shouldEqual Graph(1L)
+      Encoder[Double].apply(1.0) shouldEqual Graph(1.0)
+      Encoder[Float].apply(1.0f) shouldEqual Graph(1.0f)
+      Encoder[Boolean].apply(true) shouldEqual Graph(true)
+      Encoder[String].apply("asd") shouldEqual Graph("asd")
+    }
+    "correctly encode UUIDs" in {
+      val uuid = UUID.randomUUID()
+      Encoder[UUID].apply(uuid) shouldEqual Graph(uuid.toString)
+    }
+    "correctly encode FiniteDuration" in {
+      val duration = FiniteDuration(3, TimeUnit.DAYS)
+      Encoder[FiniteDuration].apply(duration) shouldEqual Graph(duration.toString())
+    }
+    "correctly encode Duration" in {
+      val duration = Duration.Inf
+      Encoder[Duration].apply(duration) shouldEqual Graph(duration.toString())
+    }
+    "correctly encode Instant" in {
+      val instant = Instant.now()
+      Encoder[Instant].apply(instant) shouldEqual Graph(instant.toString())
+    }
+    "correctly encode Period" in {
+      val period = Period.ofDays(3)
+      Encoder[Period].apply(period) shouldEqual Graph(period.toString())
+    }
+    "correctly encode AbsoluteIris" in {
+      Encoder[AbsoluteIri].apply(schema.value) shouldEqual Graph(schema.value)
+    }
+    "correctly encode a Set" in {
```

Comment:
On collections, maybe it would have been better to test two cases
- Sets and Lists with objects inside (that's what you tested)
```
{
  "@context": {
    "@vocab": "http://schema.org/",
   ...
  },
  "collection": [
    {
      "one": 1,
      "two": 2
    },
    {
      "one": 3,
      "two": 4
    }    
  ]
}

Triples for set:
_:b0 <http://schema.org/collection> _:b1 .
_:b0 <http://schema.org/collection> _:b2 .
_:b1 <http://schema.org/one> "1"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b1 <http://schema.org/two> "2"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b2 <http://schema.org/one> "3"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b2 <http://schema.org/two> "4"^^<http://www.w3.org/2001/XMLSchema#integer> .

Triples for list:
_:b0 <http://schema.org/collection> _:b3 .
_:b1 <http://schema.org/one> "1"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b1 <http://schema.org/two> "2"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b2 <http://schema.org/one> "3"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b2 <http://schema.org/two> "4"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b3 <http://www.w3.org/1999/02/22-rdf-syntax-ns#first> _:b1 .
_:b3 <http://www.w3.org/1999/02/22-rdf-syntax-ns#rest> _:b4 .
_:b4 <http://www.w3.org/1999/02/22-rdf-syntax-ns#first> _:b2 .
_:b4 <http://www.w3.org/1999/02/22-rdf-syntax-ns#rest> <http://www.w3.org/1999/02/22-rdf-syntax-ns#nil> .

```
- Sets and Lists with single values (that's missing)

```
{
  "@context": {
    "@vocab": "http://schema.org/",
   ...
  },
  "collection": [ 1, 2, 3, 4 ]

Triples for sets:
_:b0 <http://schema.org/collection> "1"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b0 <http://schema.org/collection> "2"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b0 <http://schema.org/collection> "3"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b0 <http://schema.org/collection> "4"^^<http://www.w3.org/2001/XMLSchema#integer> .


Triples for lists:
_:b0 <http://schema.org/collection> _:b1 .
_:b1 <http://www.w3.org/1999/02/22-rdf-syntax-ns#first> "1"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b1 <http://www.w3.org/1999/02/22-rdf-syntax-ns#rest> _:b2 .
_:b2 <http://www.w3.org/1999/02/22-rdf-syntax-ns#first> "2"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b2 <http://www.w3.org/1999/02/22-rdf-syntax-ns#rest> _:b3 .
_:b3 <http://www.w3.org/1999/02/22-rdf-syntax-ns#first> "3"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b3 <http://www.w3.org/1999/02/22-rdf-syntax-ns#rest> _:b4 .
_:b4 <http://www.w3.org/1999/02/22-rdf-syntax-ns#first> "4"^^<http://www.w3.org/2001/XMLSchema#integer> .
_:b4 <http://www.w3.org/1999/02/22-rdf-syntax-ns#rest> <http://www.w3.org/1999/02/22-rdf-syntax-ns#nil> .

}
```

Reply:
this uncovered a bug in handling set of primitives... ;)
fixed